lib/Driver/Driver.cpp

```diff
@@ -2031,6 +2031,12 @@ const ToolChain *Driver::getToolChain(const ArgList &Args) const {
       TC = new toolchains::Darwin(*this, Target);
       break;
     case llvm::Triple::Linux:
+      if (Target.isAndroid()) {
```

Comment:
I'm going to go ahead and create a class hierarchy here. There's a ton of duplication across `GenericUnix` and `Windows`, so I'll begin by submitting a separate pull request that establishes the following hierarchy:

```
GenenricUnix
    Windows
```

Then this pull request could add an additional subclass:

```
GenenricUnix
    Windows
    Android
```

Does that seem reasonable? (Admittedly, it seems strange to have `Windows` inherit from `GenericUnix`...)